### ISSUE TITLE
Extended backdrop doesn't take effect on some apps when having 6+ CPU cores

### ISSUE THREAD

**Issue by Yarrio69** [MicaForEveryone/MicaForEveryone#83]
While "Extend Frame into Client Area" is enabled, it should work as expected on several apps (any amount of cores), and Aero Wizard if we have a CPU with 4 or less cores.

But if we have a CPU with 6+ cores, the extended backdrop doesn't take effect on Aero Wizard at all until we reload MicaForEveryone config.

Examples of Aero Wizard apps are:
• Add a network location
• Bluetooth File Transfer (`fsquirt.exe`)
• Create Shortcut
• Extract Compressed (Zipped) Folders

Here's what Bluetooth File Transfer looks like on 4-core vs 6-core:
![aerowizardlight](https://user-images.githubusercontent.com/63135654/174316840-1538919a-5998-4b6a-9e61-821b3babb6be.png)
White texts displaying on white background = extended backdrop works

Another comparison when using a black theme:
![aerowizarddark](https://user-images.githubusercontent.com/63135654/174316871-583ece73-6df1-4d3c-a507-07cc2b9921fd.png)

Config:
```
Class: "NativeHWNDHost" {
  TitleBarColor = Light ; or Dark if using black theme
  BackdropPreference = Mica
  ExtendFrameIntoClientArea = True
  EnableBlurBehind = False
}
```

**Comment by xmine64** [comment]
I have 8 logical CPU cores and I had no problem. ig it's a third party incompatibility, @dongle-the-gadget what do you think?

**Comment by dongle-the-gadget** [comment]
> I have 8 logical CPU cores and I had no problem. ig it's a third party incompatibility, @dongle-the-gadget what do you think?

I don't have a machine to test it.

**Comment by xmine64** [comment]
> can you build a debug version of MFE and see if the rules apply correctly?

debug version can be grabbed from [here](https://github.com/MicaForEveryone/MicaForEveryone/actions/runs/2557780071). Still you need to attach it to a debugger to see logs. I guess it would be nice if we add a way to show logs in the UI.

**Comment by dongle-the-gadget** [comment]
Well logging functionality is perfectly doable.

**Comment by Yarrio69** [comment]
@xmine64 I think just the number of physical cores and vCPU's are affecting this, so you have 4 cores & 8 threads.
Check in Task Manager → Performance → CPU just to be sure.

My PC has an 8-core & 8-thread CPU which I have this issue, while a VM with 4 vCPU's has no problem with it (same result as yours).

I will take a look at a debug build.

**Comment by xmine64** [comment]
> I think just the number of physical cores and vCPU's are affecting this, so you have 4 cores & 8 threads.
Check in Task Manager → Performance → CPU just to be sure.

yeah I said 8 *logical* cores. so you should have 6+ physical cores to reproduce. I don't think if number of cores has any effects and I'm not sure how to fix it if it's the case.

**Comment by xmine64** [comment]
> while a VM with 4 vCPU's has no problem with it

Did you test it in a VM with 8 cores?

**Comment by Yarrio69** [comment]
Yep, setting a vCPU count to 8 also causes Mica to not work with Aero Wizard's client area.
![image](https://user-images.githubusercontent.com/63135654/188229554-16920be2-1067-4236-af35-237406c5e830.png)

**Comment by dongle-the-gadget** [comment]
I don’t have a machine with more than 4 cores to test, unfortunately.

**Comment by xmine64** [comment]
MFE is certainly working, probably it's a bug in DWM or Windows itself.

**Comment by Yarrio69** [comment]
Looks like it's a similar behaviour as File Explorer losing Mica effect after resize/unfocus.

I created an AutoHotkey script a while ago to workaround this bug
```
Loop
{
    GroupAdd, MFE, ahk_class CabinetWClass
    GroupAdd, MFE, ahk_class NativeHWNDHost
    WinWaitActive, ahk_group MFE
    FileSetTime, , %LocalAppData%\Mica For Everyone\MicaForEveryone.conf
    WinWaitNotActive
}
```
When the File Explorer or Aero Wizard window is focused, AHK updates the Date Modified timestamp of a config file so MFE automatically reloads itself and re-applies Mica effect on Client Area. "Reload on change" option must be enabled for it to work.

This not a perfect solution for Explorer as it will lose Mica effect again, so MFE needs to be reloaded by re-focusing it.

**Comment by dongle-the-gadget** [comment]
From your description, this looks like a timing bug in which MFE tries to apply extended frames too early, which would be overriden by the dialog.

We've hit this bug with Firefox before, in which a buggy implementation to enable or disable `DWMWA_USE_IMMERSIVE_DARK_MODE` would run later than MFE.

**Comment by dongle-the-gadget** [comment]
TL;DR: third party bug.

There's currently one way I could think of to somewhat mitigate this problem: defer applying rules. This approach has two problems though. One is, we don't know how long it takes between window creation and frame extending/attribute modifying, meaning there is no single "correct value" we can stick with. Secondly, (from what I know), MFE applies rules synchronously, which would add up to a very noticeable delay.

**Comment by xmine64** [comment]
> this looks like a timing bug in which MFE applied extended frames too early, which would be overriden by the dialog

This is gonna be problematic.

**Comment by ethroz** [comment]
> Another comparison when using a black theme: ![aerowizarddark](https://user-images.githubusercontent.com/63135654/174316871-583ece73-6df1-4d3c-a507-07cc2b9921fd.png)

@Yarrio69 how did you get it to look like the picture on the left? MicaForEveryone is only changing the titlebar for me, not the background of the app itself.

**Comment by Yarrio69** [comment]
@ethroz It's done by editing msstyles.

Custom themes (dark & black) can be downloaded from Rectify11 discord.

**Comment by Yarrio69** [comment]
Yeaaa, this is unfixable, as Aero Wizard window resets its Client Area backdrop when it goes to the next page. I have to manually reload MFE with AutoHotkey.

https://user-images.githubusercontent.com/63135654/190158477-c9122b65-2636-4cc7-b9d8-8e004537531d.mp4
